core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -84,6 +94,36 @@ oppia.directive('supplementalCard', [
             }
           };
 
+          $scope.consumeSolution = function() {
+            playerTranscriptService.addNewInput(
+              'Please show me the answer.', true);
+            var solution = SolutionManagerService.consumeSolution();
+            var answer = '';
+            var interactionId = stateInteractionIdService.savedMemento;
+            if (interactionId === 'GraphInput' ||
```

Comment:
It might be worth moving some of the more complicated logic into some kind of solution service that can be shared between the editor and the player. Remember how we refactored hints.